aws/cloudfront_distribution_configuration_structure.go

```diff
@@ -256,14 +265,72 @@ func expandCacheBehaviors(s *schema.Set) *cloudfront.CacheBehaviors {
 	}
 }
 
-func flattenCacheBehaviors(cbs *cloudfront.CacheBehaviors) *schema.Set {
+func flattenCacheBehaviorsDeprecated(cbs *cloudfront.CacheBehaviors) *schema.Set {
 	s := []interface{}{}
 	for _, v := range cbs.Items {
-		s = append(s, flattenCacheBehavior(v))
+		s = append(s, flattenCacheBehaviorDeprecated(v))
 	}
 	return schema.NewSet(cacheBehaviorHash, s)
 }
 
+func expandCacheBehaviors(lst []interface{}) *cloudfront.CacheBehaviors {
+	var qty int64
+	var items []*cloudfront.CacheBehavior
+	for _, v := range lst {
+		items = append(items, expandCacheBehavior(v.(map[string]interface{})))
+		qty++
+	}
+	return &cloudfront.CacheBehaviors{
+		Quantity: aws.Int64(qty),
+		Items:    items,
+	}
+}
+
+func flattenCacheBehaviors(cbs *cloudfront.CacheBehaviors) []interface{} {
+	lst := []interface{}{}
+	for _, v := range cbs.Items {
+		lst = append(lst, flattenCacheBehavior(v))
+	}
+	return lst
+}
+
+// Deprecated.
+func expandCacheBehaviorDeprecated(m map[string]interface{}) *cloudfront.CacheBehavior {
+	cb := &cloudfront.CacheBehavior{
+		Compress:             aws.Bool(m["compress"].(bool)),
+		ViewerProtocolPolicy: aws.String(m["viewer_protocol_policy"].(string)),
+		TargetOriginId:       aws.String(m["target_origin_id"].(string)),
+		ForwardedValues:      expandForwardedValues(m["forwarded_values"].(*schema.Set).List()[0].(map[string]interface{})),
+		DefaultTTL:           aws.Int64(int64(m["default_ttl"].(int))),
+		MaxTTL:               aws.Int64(int64(m["max_ttl"].(int))),
+		MinTTL:               aws.Int64(int64(m["min_ttl"].(int))),
+	}
+
+	if v, ok := m["trusted_signers"]; ok {
+		cb.TrustedSigners = expandTrustedSigners(v.([]interface{}))
+	} else {
+		cb.TrustedSigners = expandTrustedSigners([]interface{}{})
+	}
+
+	if v, ok := m["lambda_function_association"]; ok {
+		cb.LambdaFunctionAssociations = expandLambdaFunctionAssociations(v.(*schema.Set).List())
+	}
+
+	if v, ok := m["smooth_streaming"]; ok {
+		cb.SmoothStreaming = aws.Bool(v.(bool))
+	}
+	if v, ok := m["allowed_methods"]; ok {
+		cb.AllowedMethods = expandAllowedMethodsDeprecated(v.([]interface{}))
```

Comment:
This also has the old behavior. The new behavior is a Set rather than lists for allowed_methods & cached_method